test/testsyncmove.cpp

```diff
@@ -857,6 +857,58 @@ private slots:
         QVERIFY(fakeFolder.syncOnce());
         QCOMPARE(fakeFolder.currentLocalState(), fakeFolder.currentRemoteState());
     }
+
+    void testMovedWithError()
+    {
+#define VFS 1
+        const auto getName = [] (const QString &s)
+        {
+#if VFS
+            return QStringLiteral("%1.owncloud").arg(s);
+#else
+            return s;
+#endif
+        };
+        const QByteArray testPath = "folder/folderA/file.txt";
+        FakeFolder fakeFolder{ FileInfo{ QString(), { FileInfo{ QStringLiteral("folder"), { FileInfo{ QStringLiteral("folderA"), { { QStringLiteral("file.txt"), 400 } } }, QStringLiteral("folderB") } } } } };
+
+        QCOMPARE(fakeFolder.currentLocalState(), fakeFolder.currentRemoteState());
+
+#if VFS
+        auto vfs = QSharedPointer<Vfs>(createVfsFromPlugin(Vfs::WithSuffix).release());
+        fakeFolder.switchToVfs(vfs);
+        fakeFolder.syncJournal().internalPinStates().setForPath("", PinState::OnlineOnly);
+        auto state = fakeFolder.syncJournal().internalPinStates().effectiveForPath(testPath);
+        QVERIFY(state);
+        QCOMPARE(*state, PinState::OnlineOnly);
+#endif
+
+        // make files virtual
+        fakeFolder.syncOnce();
+
+        fakeFolder.serverErrorPaths().append(testPath, 403);
+        fakeFolder.localModifier().rename(getName(testPath), getName("folder/folderB/file.txt"));
+
+        // sync1 file gets detected as error, instruction is still NEW_FILE
+        fakeFolder.syncOnce();
+
+        // sync2 file is in error state, checkErrorBlacklisting sets instruction to IGNORED
+        fakeFolder.syncOnce();
+
+#if VFS
+        // TODO: why does the comparison need local files
+        // comparison of states fails as local files have a suffix
```

Comment:
I guess that's because the local state has the suffix and the remote test doesn't.
Maybe this was just not implemented in the test framework.  Check if the suffix virtual file test do the comparison.